.github/workflows/lint-check-types-and-deploy.yml

```diff
@@ -0,0 +1,43 @@
+name: Linting and type checking
+on: push
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    env:
+      PGHOST: localhost
+      PGDATABASE: next_js_example_spring_2023
+      PGUSERNAME: next_js_example_spring_2023
+      PGPASSWORD: next_js_example_spring_2023
+      NEXT_PUBLIC_GIT_HUB_API: xxxxxxxxxxxxxxxxx
+    steps:
+      - name: Start preinstalled PostgreSQL on Ubuntu
+        run: |
+          sudo systemctl start postgresql.service
+          pg_isready
+      - name: Create database user
+        run: |
+          sudo -u postgres psql --command="CREATE USER next_js_example_spring_2023 PASSWORD 'next_js_example_spring_2023'" --command="\du"
+      - name: Create database and allow user
+        run: |
+          sudo -u postgres createdb --owner=next_js_example_spring_2023 next_js_example_spring_2023
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: '8'
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Install SafeQL if not yet installed (eg. on Windows dev machines)
+        run: grep package.json -e '@ts-safeql/eslint-plugin' || pnpm add @ts-safeql/eslint-plugin libpg-query
+      - run: pnpm migrate up
+      - name: Run TypeScript Compiler
+        run: pnpm tsc
+      - name: Run ESLint
+        run: pnpm eslint . --max-warnings 0
+      - name: Run linter
+        run: pnpm lint
```

Comment:
what is this?

Reply:
https://github.com/upleveled/graphql-example-spring-2023-austria-vienna/pull/3/commits/cdb0582b708375b326a18c60d2a4feca639d4b9a